### DISCUSSION TITLE
State-of-the-art for Time series sensors data generation

### DISCUSSION THREAD

**Discussion by alevangel** [synthetichealth/synthea#1284] [General]
What is the current state of the art for generating sensor data that are consistent with the patient's clinical life generated by synthea?

Are there any libraries available?

**Comment by awatson1978** [comment]
Hello,
State of the art is in pull request https://github.com/synthetichealth/synthea/pull/1254.  

A few things to consider:
- Sensors often output data in assembly/hardware language and proprietary data formats.
- Sensors rarely output data in formats that are directly consumable by hospital EHRs.  
- Sensors are also subject to data collection errors (i.e. mispositioned, poor signal quality, etc).

What we currently have staged (and available for state-of-the-art testing, for those who are brave enough to try), are physiology models from the [BioModel](https://www.ebi.ac.uk/biomodels/) database.  These models use various partial differential equations to generate the physiology data (unlike the markov random number generators used in other parts of Synthea). The physiology data can then be mapped into either standard FHIR format, or some sort of proprietary data format (such as CSV or XML).  

At the current time, we're going to be reluctant at implementing hardware level sensor data in Synthea, as following that white rabbit leads to having to implement different output formats for however many medical devices are on the market.  Rather, the approach we are currently looking at is defining a virtual sensor device that uses FHIR as it's output format, and going directly from physiology model to standardized interoperability formats.  If somebody wants a data generator that exports in raw hardware format, they can map from FHIR to the raw data model.  

We do have a sensor data mapping utility in the works, which maps data from Apple Health and Garmin and other devices into 
 FHIR format; and which we are currently in the process of working through our publication system.  Once released, that utility would be an appropriate place to add mappings from the virtual sensor device to actual hardware data models (or vice-versa).
